### PR TITLE
#7037 Fix group title returning as undefined even though it's present

### PR DESCRIPTION
**Pull request by tonym3nsa**
## Description

Fix group titles that wrongly display their respective group names after a context page is refreshed
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
**What is the current behavior?**

#7037 

**What is the new behavior?**

Group titles are displayed instead of the group names after the context page is refreshed
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — APPROVED** [review]

File: web/client/utils/LayersUtils.js [hunk 1]
```diff
@@ -150,10 +150,42 @@ export const getNestedGroupTitle = (id, groups = []) => {
         return nodeObj ? nodeObj.title : null;
     }));
 };
+
+/**
+ * Flatten nested array to a one-level array
+ * @param {array} array of objects
```
Comment (MV88) [inline web/client/utils/LayersUtils.js:156]: if you have to uses arrays of object as type you have to do like this
```suggestion
 * @param {object[]} array of objects
```

File: web/client/utils/LayersUtils.js [hunk 2]
```diff
@@ -150,10 +150,42 @@ export const getNestedGroupTitle = (id, groups = []) => {
...
+ * @returns {array} flattened array
```
Comment (MV88) [inline web/client/utils/LayersUtils.js:157]:
```suggestion
 * @returns {object[]} flattened array
```

File: web/client/utils/LayersUtils.js [hunk 3]
```diff
@@ -150,10 +150,42 @@ export const getNestedGroupTitle = (id, groups = []) => {
...
+            result = result.concat(flattenArrayOfObjects(a.nodes));
+        }
+    });
+    return result;
+};
+
+/**
+ * Gets group title by id
+ * @param {string} id of group
+ * @param {array} groups groups of map
```
Comment (MV88) [inline web/client/utils/LayersUtils.js:173]:
```suggestion
 * @param {object[]} groups groups of map
```

**Review by MV88 — CHANGES_REQUESTED** [review]

**Review by MV88 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
Test passed, @tonym3nsa please backport this to 2021.02.xx

**Pull request merged**